internal/tester_definition.go

```diff
@@ -18,12 +18,76 @@ var testerDefinition = testerutils.TesterDefinition{
 	ExecutableFileName: "your_server.sh",
 	Stages: []testerutils.Stage{
 		{
-			Slug:                    "init",
+			Slug:                    "stage-1",
 			Number:                  1,
+			Title:                   "Can connect to a TCP server",
+			TestFunc:                testConnects,
+			ShouldRunPreviousStages: true,
+			Timeout:                 15 * time.Second,
+		},
+		{
+			Slug:                    "stage-2",
+			Number:                  2,
 			Title:                   "Respond with 200",
 			TestFunc:                test200OK,
 			ShouldRunPreviousStages: true,
 			Timeout:                 15 * time.Second,
 		},
+		{
+			Slug:                    "stage-3",
+			Number:                  3,
+			Title:                   "Respond with 404",
+			TestFunc:                test404NotFound,
+			ShouldRunPreviousStages: true,
+			Timeout:                 15 * time.Second,
+		},
+		{
+			Slug:                    "stage-4",
+			Number:                  4,
+			Title:                   "Respond with content",
+			TestFunc:                testRespondWithContent,
+			ShouldRunPreviousStages: true,
+			Timeout:                 15 * time.Second,
+		},
+		{
+			Slug:                    "stage-5",
+			Number:                  5,
+			Title:                   "Header Parsing",
+			TestFunc:                testRespondWithUserAgent,
+			ShouldRunPreviousStages: true,
+			Timeout:                 15 * time.Second,
+		},
+		{
+			Slug:                    "stage-6",
```

Comment:
@sherubthakur I'd name these with a slug that maps to the title - sometimes based on user feedback we end up switching the order of stages, and slugs help us map to the correct stage after the order has changed.